conifer/model.py

```diff
@@ -350,30 +355,120 @@ def build(self, **kwargs):
         '''
         raise NotImplementedError
 
-    def profile(self, bins=50, return_data=False, return_figure=True):
+    def profile(self, what, ax=None):
+        """Profile the thresholds or gains of the trees in the ensemble for each feature.
+        
+        For each feature:
+        - Orange line: median
+        - Box plot: First-third quantile (25%,75%)
+        - Black caps: 2.5% and 97.5% percentiles
+        - Gray caps: Full range
+
+        Args:
+            what ("gains"|"thrs"): choose between profiling the thresholds or the gains.
+            ax (matplotlib.axes._axes.Axes, optional): Matplotlib axes. Defaults to None.
+
+        Returns:
+            matplotlib.axes._axes.Axes: Matplotlib axes.
+        """
         try:
             import matplotlib.pyplot as plt
+            import mplhep
         except ImportError:
-            raise Exception("matplotlib not found. Please install matplotlib")
-        value = np.concatenate([np.array(tree.value) for trees in self.trees for tree in trees])
-        threshold = np.concatenate([np.array(tree.threshold) for trees in self.trees for tree in trees])
-        hv, bv = np.histogram(value, bins=bins)
-        wv = bv[1] - bv[0]
-        ht, bt = np.histogram(threshold, bins=bins)
-        wt = bt[1] - bt[0]
-        figure = plt.figure()
-        plt.subplot(1, 2, 1)
-        plt.bar(bv[:-1]+wv/2, hv, width=wv)
-        plt.xlabel('Distribution of tree values (scores)')
-        plt.subplot(1, 2, 2)
-        plt.bar(bt[:-1]+wt/2, ht, width=wt)
-        plt.xlabel('Distribution of tree thresholds')
-        if return_data and return_figure:
-            return (value, threshold, figure)
-        elif return_data:
-            return (value, threshold)
-        elif return_figure:
-            return (figure)
+            raise Exception("matplotlib or mplhep not found. Please install matplotlib and mplhep")
+        mplhep.style.use("CMS")
```

Comment:
I would prefer not to use this styling in the `conifer` tool itself. Of course if one wants the CMS styling they can run this command in their own script

Reply:
I have removed the mplhep dependency and replaced by a minimal set of style options (taken from mplhep) to make the plot look better